test/library.cpp

```diff
@@ -236,14 +236,14 @@ namespace
 
 TEST(LibraryOpdsImportTest, allInOne)
 {
-  kiwix::Library lib;
-  kiwix::Manager manager(&lib);
+  auto lib = std::make_shared<kiwix::Library>();
+  kiwix::Manager manager(lib);
   manager.readOpds(sampleOpdsStream, "library-opds-import.unittests.dev");
 
-  EXPECT_EQ(10U, lib.getBookCount(true, true));
+  EXPECT_EQ(10U, lib->getBookCount(true, true));
 
   {
-  const kiwix::Book& book1 = lib.getBookById("0c45160e-f917-760a-9159-dfe3c53cdcdd");
+  const kiwix::Book& book1 = lib->getBookById("0c45160e-f917-760a-9159-dfe3c53cdcdd");
```

Comment:
You could have avoided the numerous changes like this one by utilizing a `shared_ptr` with a non-deleting deallocator:

https://github.com/kiwix/libkiwix/blob/a844bc4000de0cce1e0b5a1b0b5356a3160fd99a/src/manager.cpp#L33-L36

https://github.com/kiwix/libkiwix/blob/a844bc4000de0cce1e0b5a1b0b5356a3160fd99a/src/manager.cpp#L92-L102